libs/watt/src/lib/components/picker/datepicker/watt-datepicker.component.ts

```diff
@@ -205,6 +207,25 @@ export class WattDatepickerComponent extends WattPickerBase {
         this.datePlaceholder = this.getPlaceholderByLocale(locale);
         this.rangePlaceholder = this.getRangePlaceholder();
       });
+
+    effect(() => {
+      this.rangeMonthOnlyMode();
+      this.ngControl?.control?.updateValueAndValidity();
+    });
+  }
+
+  override ngAfterViewInit() {
+    this.ngControl?.control?.setValidators(this.validate.bind(this));
+  }
+
+  validate({ value }: AbstractControl<WattRange<string>>) {
+    if (!value?.end || !value?.start) return null;
+    if (!this.rangeMonthOnlyMode()) return null;
+    const start = dayjs(value.start);
+    const end = dayjs(value.end);
+    return start.isSame(start.startOf('month')) && end.isSame(start.endOf('month'))
+      ? null
+      : { monthOnly: true };
```

Comment:
issue: looks like this causes some issues. The most obvious I can see is the required indicator next to the label.

Before

![image](https://github.com/user-attachments/assets/772e47af-78aa-4c17-9f0a-b417e1a2dd27)

After

![image](https://github.com/user-attachments/assets/6ee9ff87-75c7-494e-b0fc-b0f84cf50d06)

I think the `null` return at the start of the method overwrites the other validation errors

Reply:
Thanks for trying it out, well spotted! I think the `null` return is fine, since that is the way for validators to tell that there is no error. I traced the issue to be with the `setValidators` call in `ngAfterViewInit`. I initially thought that the `setValidators` would be called before other validators are set, but that was wrong, since the form + validators are always instantiated in the parent component before child components render. I fixed it be changing it to a "addValidators" call instead.

0ab3309b636065fff7f13b5f57507984f017956e